### PR TITLE
Add new hostClientTypes

### PR DESCRIPTION
**Pull request by jmsy**
Add new hostClientType for Teams Android devices and MTR on Windows devices.

### CODE REVIEW

File: src/public/interfaces.ts [hunk 1]
```diff
@@ -334,7 +334,8 @@ export interface Context {
   isTeamArchived?: boolean;
 
   /**
-   * The type of the host client. Possible values are : android, ios, web, desktop, rigel
+   * The type of the host client. Possible values are : android, ios, web, desktop, rigel,
```
Comment (steffenloesch) [inline src/public/interfaces.ts:337]: I think it would be good to indicate that rigel is being deprecated.

Comment (jmsy) [reply src/public/interfaces.ts]: Thanks, added @deprecated annotation.

**Review by ydogandjiev — APPROVED** [review]

File: src/private/privateAPIs.ts [hunk 2]
```diff
@@ -35,7 +35,10 @@ export function getUserJoinedTeams(
   ensureInitialized();
 
   if (
-    GlobalVars.hostClientType === HostClientType.android &&
+    (GlobalVars.hostClientType === HostClientType.android ||
+      GlobalVars.hostClientType === HostClientType.teamsRoomsAndroid ||
```
Comment (ydogandjiev) [inline src/private/privateAPIs.ts:39]: Why add these here? This change was made to address a very specific issue in Android Mobile Client. I don't think we should be adding more entries here.

**Review by ydogandjiev — CHANGES_REQUESTED** [review]

File: src/private/privateAPIs.ts (hunk 2 above)
Comment (jmsy) [reply src/private/privateAPIs.ts:39]: Originally all the Android platforms, including mobile and devices were sharing the same HostClientType - `android`, so I assume here the we should apply the logic to all the Android platforms to keep the same behavior as before. Do you think it's unnecessary? How about other changes in `authentication.ts` and `media.ts` ?

Comment (ydogandjiev) [reply src/private/privateAPIs.ts:39]: Hmm... Are you saying that all of these are just variations of our Android client? Do you know who owns the Client SDK implementation for these different client types? @shbhadra, can you please chime in here.

Comment (shbhadra) [reply src/private/privateAPIs.ts:39]: Hey Edward, is the corresponding change for this checked into the android client? 
The value for `GlobalVars.hostClientType` is set during the initialize() call, and is a value that is sent by the client. 
Android code seems to be still setting this as 'android' always.

Comment (jmsy) [reply src/private/privateAPIs.ts:39]: I have another PR to make the hostClientType change on Android side, it's not merged to develop now. All the new types here are subtype of "android". 
https://domoreexp.visualstudio.com/Teamspace/_git/SkypeSpaces-Android/pullrequest/352658

**Review by ydogandjiev — APPROVED** [review]

**Review by shbhadra — APPROVED** [review]

**Pull request merged**